tests/characters/test_bob_handles_frags.py

```diff
@@ -61,6 +61,9 @@ def test_bob_can_issue_a_work_order_to_a_specific_ursula(enacted_policy, alice,
     networky_stuff = MockNetworkyStuff(ursulas)
     ursula_dht_key, work_order = list(work_orders.items())[0]
 
+    # In the real world, we'll have a full Ursula node here.  But in this case, we need to fake it.
+    work_order.ursula = ursulas[0]
```

Comment:
ursulas[0] isn't necessarily the one corresponding to ursula_dht_key
So at the end, bob._saved_work_orders has both ursula_dht_key and ursulas[0].interface_dht_key() if they are different

Reply:
Fix:
```
    for ursula in ursulas:
        if ursula.interface_dht_key() == ursula_dht_key:
            work_order.ursula = ursula
            break
```

Reply:
Fixed instead by implementing an `OrderedDict` both going into and coming out of the `TreasureMap`.